### DISCUSSION TITLE
Plane2Cylendar Perspective

### DISCUSSION THREAD

**Discussion by LukePittman** [ImageMagick/ImageMagick#3085] [Digital Image Processing]
Hey everyone!  Trying to solve this problem, but really need help.

I am using Plane2Cylendar to 'wrap' an image around the subject in another image.  It's currently working great, except I need to add a perspective to it, like the eye level of the user was actually above the center of the image being wrapped.  To help make more sense of that, I've created this illustration:

![lprmI](https://user-images.githubusercontent.com/38956230/103834034-2d65c600-5037-11eb-8743-6dd10a4ef0bf.png)

Note that I am doing this from PHP and not command-line, but am happy to work on it via command line if needed.

`$label = new Imagick ($label);
$label->setImageBackgroundColor(new \ImagickPixel('transparent'));
$label->setImageVirtualPixelMethod(Imagick::VIRTUALPIXELMETHOD_TRANSPARENT);
$label->distortImage(\Imagick::DISTORTION_PLANE2CYLINDER, [27], true);
`

I then overlay $label on my main image and output it.

Hopefully that's enough info.  Would really love some help.  Happy to pay too.

**Comment by fmw42** [comment]
Have you tried -distort barrel?  See https://legacy.imagemagick.org/Usage/distorts/#barrel. Also see -compose displace example at https://legacy.imagemagick.org/Usage/mapping/#displace_cylinder.

Or if you are on a Unix-like platform (Linux, Mac OSX, Window 10 Unix), then try my cylinderize script from my website at http://www.fmwconcepts.com/imagemagick

**Reply by LukePittman** [reply]
The "Fred"!  I've been reading through your pages for awhile trying to get my head around it.  You're work is great, however much of it is beyond me.  Are you available for hire?

**Reply by fmw42** [reply]
Contact me offline. My email is the licensing section just under the list of scripts at the top of my home page.
